apps/web/components/thread/citation-provider.tsx

```
@@ -0,0 +1,109 @@
+
+      const newCitations = uniqueCitations.reduce((citations, url, index) => {
+        console.log(url);
+        const isValid = isValidUrl(url);
+        const host = getHost(url);
+        if (!host) return citations;
+        const favIcon = getFavIcon(host);
+        if (!isValid) return citations;
+
+        if (index < MAX_VISIBLE_CITATIONS) {
+          return {
+            ...citations,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)